api/ingest.py

```diff
@@ -14,7 +14,7 @@
 
 
 class IngestAPI:
-    def __init__(self, url=None):
+    def __init__(self, url):
```

Comment:
Is it safe to make url a required parameter i.e all code that calls IngestAPI() passes in the required `url` param?

Reply:
I've put it back now. 👍  source it from param or from config/env var.

Although i think it's fine to explicitly say this is the url i want to connect to when instantiating an IngestAPI object.

Reply:
> Although i think it's fine to explicitly say this is the url i want to connect to when instantiating an IngestAPI object.
Yeah, personally I think that's how it should always be initialised, maybe then with a static `IngestAPI.with_env_defaults()` if we want to handoff construction to env vars